src/tiffcomposite_int.hpp

```diff
@@ -11,1561 +11,1658 @@
+        template <const ArraySet* arraySet, int N, CfgSelFct cfgSelFct>
+        TiffComponent::UniquePtr newTiffBinaryArray2(uint16_t tag, IfdId group)
+        {
+            return TiffComponent::UniquePtr(new TiffBinaryArray(tag, group, arraySet, N, cfgSelFct));
+        }
 
-    //! Function to create and initialize a new TIFF entry for a thumbnail (data)
-    template<uint16_t szTag, IfdId szGroup>
-    TiffComponent::UniquePtr newTiffThumbData(uint16_t tag, IfdId group)
-    {
-        return std::make_unique<TiffDataEntry>(tag, group, szTag, szGroup);
-    }
+        //! Function to create and initialize a new TIFF entry for a thumbnail (data)
+        template <uint16_t szTag, IfdId szGroup>
+        TiffComponent::UniquePtr newTiffThumbData(uint16_t tag, IfdId group)
+        {
+            return std::make_unique<TiffDataEntry>(tag, group, szTag, szGroup);
+        }
 
-    //! Function to create and initialize a new TIFF entry for a thumbnail (size)
-    template<uint16_t dtTag, IfdId dtGroup>
-    TiffComponent::UniquePtr newTiffThumbSize(uint16_t tag, IfdId group)
-    {
-        return std::make_unique<TiffSizeEntry>(tag, group, dtTag, dtGroup);
-    }
+        //! Function to create and initialize a new TIFF entry for a thumbnail (size)
+        template <uint16_t dtTag, IfdId dtGroup>
+        TiffComponent::UniquePtr newTiffThumbSize(uint16_t tag, IfdId group)
+        {
+            return std::make_unique<TiffSizeEntry>(tag, group, dtTag, dtGroup);
+        }
 
-    //! Function to create and initialize a new TIFF entry for image data
-    template<uint16_t szTag, IfdId szGroup>
-    TiffComponent::UniquePtr newTiffImageData(uint16_t tag, IfdId group)
-    {
-        return std::make_unique<TiffImageEntry>(tag, group, szTag, szGroup);
-    }
+        //! Function to create and initialize a new TIFF entry for image data
+        template <uint16_t szTag, IfdId szGroup>
+        TiffComponent::UniquePtr newTiffImageData(uint16_t tag, IfdId group)
+        {
+            return std::make_unique<TiffImageEntry>(tag, group, szTag, szGroup);
+        }
 
-    //! Function to create and initialize a new TIFF entry for image data (size)
-    template<uint16_t dtTag, IfdId dtGroup>
-    TiffComponent::UniquePtr newTiffImageSize(uint16_t tag, IfdId group)
-    {
-        // Todo: Same as newTiffThumbSize - consolidate (rename)?
-        return std::make_unique<TiffSizeEntry>(tag, group, dtTag, dtGroup);
-    }
+        //! Function to create and initialize a new TIFF entry for image data (size)
+        template <uint16_t dtTag, IfdId dtGroup>
+        TiffComponent::UniquePtr newTiffImageSize(uint16_t tag, IfdId group)
+        {
+            // Todo: Same as newTiffThumbSize - consolidate (rename)?
+            return std::make_unique<TiffSizeEntry>(tag, group, dtTag, dtGroup);
+        }
 
-}}                                      // namespace Internal, Exiv2
+    }  // namespace Internal
```

Comment:
This feels like it should be merged. I couldn't get it to compile afterwards though.

Reply:
I do not think clang-format can do this automatically. Imagine the following case:

```cpp
namespace a
{
  namespace b
  {
  }
  namespace c
  {
  }
}
```
It would be wrong for clang-format to try to attempt such conversions.

Please take into account that I do not want to make manual modifications in this PR. The changes applied by clang-format are already too large to add more manual changes here.